### PR TITLE
Add CPUOffload

### PR DESCRIPTION
**Pull request by apaz-cli**
Still have to test MFU, was slower ofc b/c will do so once out of hackathon.

I also tested CPUOptimizer last night, but it's roughly on par with torch cpu adam when it isn't pipelined, and there's still the gradient clipping problem. So leaving it out for simplicity's sake. If we solved the gradient clipping problem though there would be a huge speedup from adding it.

### CODE REVIEW

**Comment by apaz-cli** [comment]
`uv run torchrun --nproc_per_node=8 src/zeroband/train.py @ configs/training/Qwen7B.toml --data.fake --train.micro_bs 1 --optim.total_steps 2 --train.ac_ckpt true --train.attn_impl flash_attention_2 --train.liger_qwen true`
```hs
03:34:25 [INFO] [TRAIN] [Rank 0] step: 2, loss: 0.3170, average_rewards: -0.0055, tokens_per_second: 68076.18, mfu: 44.99
03:34:25 [INFO] [TRAIN] [Rank 0] Max memory: 35.98 GB
```

`uv run torchrun --nproc_per_node=8 src/zeroband/train.py @ configs/training/Qwen7B.toml --data.fake --train.micro_bs 1 --optim.total_steps 2 --train.ac_ckpt true --train.attn_impl flash_attention_2 --train.liger_qwen true --train.fsdp_cpuoffload true`
```hs
04:25:24 [INFO] [TRAIN] [Rank 0] step: 2, loss: 0.3170, average_rewards: -0.0055, tokens_per_second: 44456.64, mfu: 29.38
04:25:24 [INFO] [TRAIN] [Rank 0] Max memory: 25.35 GB
```
